syntax = "proto3";

import "google/protobuf/empty.proto";

option go_package = ".;proto";

// 商品分类
service Category {
    // 获取所有的分类
    rpc GetAllCategoryList (google.protobuf.Empty) returns (CategoryListResponse);
    // 获取子分类
    rpc GetSubCategory (CategoryListRequest) returns (SubCategoryListResponse);
    //新建分类
    rpc CreateCategory (CategoryInfoRequest) returns (CategoryInfoResponse);
    // 删除分类
    rpc DeleteCategory (DeleteCategoryInfo) returns (google.protobuf.Empty);
    // 修改分类
    rpc UpdateCategory (CategoryInfoRequest) returns (google.protobuf.Empty);

}

message CategoryFilterFilterRequest {
    int32 id = 1;
    bool is_tab = 2;
}
message CategoryListResponse {
    int32 total = 1;
    repeated CategoryInfoResponse data = 2;
    string json_data = 3;
}
message CategoryListRequest {
    int32 id = 1;
    int32 level = 2;
}
message SubCategoryListResponse {
    int32 total = 1;
    CategoryInfoResponse info = 2;
    repeated CategoryInfoResponse sub_category_list = 3;
}


message CategoryInfoResponse {
    int32 id = 1;
    string name = 2;
    int32 parent_category = 3;
    int32 level = 4;
    bool is_tab = 5;
}
message DeleteCategoryInfo {
    int32 id = 1;
}

message QueryCategoryRequest {
    int32 id = 1;
    string name = 2;
};

message BatchCategoryInfoRequest {
    repeated int32 id = 1;
    int32 goods_nums = 2;
    int32 brand_nums = 3;
}


message CategoryInfoRequest {
    int32 id = 1;
    string name = 2;
    int32 parent_category = 3;
    int32 level = 4;
    bool is_tab = 5;
}
